### PR TITLE
Added resequence(), recollect() and qpackage() functions

### LINKED ISSUE #23: Add resequence () function to qDatr
The function sorts and filters dates.

### LINKED ISSUE #25: Create qpackage_create()

### LINKED ISSUE #26: Add interleave () function
For resetting the century of observations with unlikely future events.

### LINKED ISSUE #30: Add recollect () function
Pastes together unique rows/observations.

### PR DESCRIPTION
**Pull request by henriquesposito**
# Description

## Data wrangling

 * Closes #26 by adding `interleave()`
 * Closes #23 by adding `resequence()`
 * Closes #30 by adding `recollect()`

 ## Data packages

 * Closes #25 by adding `create_qpackage()`
 * Fixed 'pushrelease.yml' postfix bug

# Checklist:

- [x] I have commented my code in-line, particularly in hard-to-understand areas
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have documented any functions or data in the corresponding .R scripts so that roxygen can create documentation
- [x] The package builds on my OS without issues
- [x] My changes generate no new warnings
- [x] PR checks all pass for latest commit
- [x] I have bumped the version by the appropriate increment in the DESCRIPTION file (major, minor, patch)
- [x] I have updated NEWS.md
- [x] I have given this pull request an informative title and description

### CODE REVIEW

**Comment by henriquesposito** [comment]
The error I get when running 'devtools::check' in my computer is: " Namespace dependency not required: 'usethis' ".  The package 'usethis' is listed as a dependency under namespace and not in the description file or anywhere else in the package that I can find. Why is this happening?

**Comment by jhollway** [comment]
We will be using `usethis` for this function, but it is not yet written into the function. You can comment out the @import line from the R script for now, which should remove it from NAMESPACE when you rerun documentation.

File: R/package.R [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#' Create a new package in the qDatr ecosystem
+#'
+#' Creates a new package in and consistent with the qDatr ecosystem
+#' @param packageName A string giving the desired name of the package, must start with "q"
```
Comment (jhollway) [inline R/package.R:4]: Line over 80 characters

File: R/package.R [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#' @param packageAuthor A string, list or vector giving the package author(s), required
```
Comment (jhollway) [inline R/package.R:5]: Line over 80 characters

File: R/package.R [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+#' @param update A logical indicating whether existing files should be overwritten, by default TRUE.
```
Comment (jhollway) [inline R/package.R:6]: Line over 80 characters

File: R/package.R [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+#' TODO: do these checks internally and smartly...
+#' @details The function establishes many of the required files and folder structures
```
Comment (jhollway) [inline R/package.R:8]: Line over 80 characters

File: R/package.R [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+#' required for a qDatr-consistent data package.
+#' @return A new package structure
```
Comment (jhollway) [inline R/package.R:10]: Remove trailing whitespace

File: R/package.R [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+#' @return A new package structure 
+#' @importFrom usethis create_tidy_package
+#' @examples
+#' \dontrun{
+#' qpackage_create("qStates", c("James Hollway", "Henrique Sposito"))
+#' }
+#' @export
+create_qpackage <- function(packageName, 
+                            packageAuthor,
+                            update = TRUE){
```
Comment (jhollway) [inline R/package.R:19]: Add space between right parenthesis and opening curly brace

File: R/package.R [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  
+  if(is.null(packageName)) stop("Please declare a package name")
+  if(!startsWith(packageName, "q")) stop("Package name must start with a 'q'")
+  if(is.null(packageAuthor)) stop("Please declare at least one author")
```
Comment (jhollway) [inline R/package.R:23]: Cover these lines with tests

File: R/standardise.R [hunk 8]
```diff
@@ -49,3 +49,48 @@ entitle <- function(s, strict = FALSE) {
...
+
+  out <- apply(data[,vars], 1, function(x) {
+    dates <- sort(unlist(strsplit(unique(na.omit(x)),unity)))
+
+    if (length(dates) < len){
+      dates <- interleave(dates, which(is.na(x)))
+    }
+
+    if (length(dates) > len){
+      if (sum((!grepl("-01-01", dates))*1)>=len) dates <- dates[!grepl("-01-01",dates)]
```
Comment (jhollway) [inline R/standardise.R:82]: Line not covered by tests?

**Review by jhollway — CHANGES_REQUESTED** [review]

**Review by jhollway — APPROVED** [review]
Looks good, thanks @henriquesposito

**Pull request merged**